cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_resources.go

```diff
@@ -458,6 +441,15 @@ func (s *Server) inspectKappK8sResources(ctx context.Context, cluster, namespace
 	// Fetch the GroupVersions used by the app
 	usedGVs, err := app.UsedGVs()
 	if err != nil {
+		// TODO(minelson): We can't currently use `errors.IsNotFound(err)` here.
+		// See https://github.com/vmware-tanzu/carvel-kapp/issues/498
+		// Instead we need to match on the error string :/
+		if strings.Contains(err.Error(), fmt.Sprintf("configmaps %q not found", appName)) {
+			// We want to return a NotFound here because the dashboard already
+			// handles this case, knowing that the references may not be
+			// available immediately.
+			return nil, status.Errorf(codes.NotFound, "App not found: %+v", err)
```

Comment:
Besides, perhaps a `NotFound`error perhaps is not the best fit, I'd rather use the `UNAVAILABLE` code, whose semantics is more aligned to sth like "ey client, right now I don't have it, but you can wait with some backoff if you want"